pyomo/contrib/pynumero/interfaces/external_pyomo_model.py

```diff
@@ -276,74 +223,27 @@ def equality_constraint_names(self):
         return ["residual_%i" % i for i in range(self.n_equality_constraints())]
 
     def set_input_values(self, input_values):
+        self._timer.start("set_inputs")
```

Comment:
My motivation in the timing calls is so that timing statistics like this:
```console
Identifier                 ncalls   cumtime   percall      %
------------------------------------------------------------
root                            1    11.990    11.990  100.0
     -------------------------------------------------------
     __init__                  10     4.688     0.469   39.1
               ---------------------------------------------
               PyomoNLP        10     2.881     0.288   61.5
               other          n/a     1.807       n/a   38.5
               =============================================
     hessian                  100     1.447     0.014   12.1
     jacobian                 110     0.625     0.006    5.2
     set_inputs               130     4.572     0.035   38.1
               ---------------------------------------------
               solve_nlp    48230     4.049     0.000   88.6
               other          n/a     0.522       n/a   11.4
               =============================================
     other                    n/a     0.658       n/a    5.5
     =======================================================
============================================================
```
can be generated without intrusive modification. I think for time-consuming methods that we might commonly want to profile, the timing calls are justified. See above for a performance diff when timing is introduced in small, fast methods that are called often, which suggests that they should not be a permanent feature of these methods.